File: src/main/java/com/fifo/ticketing/domain/performance/dto/AdminPerformanceBookDetailDto.java
package com.fifo.ticketing.domain.performance.dto;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@RequiredArgsConstructor
public class AdminPerformanceBookDetailDto {

    private final Long id;
    private final String title;
    private final String encodedFileName;
    private final Long totalPrice;
    private final Long totalQuantity;

    @Setter
    private String urlPrefix;

    public String getUrl() {
        return urlPrefix + encodedFileName;
    }

}

File: src/main/java/com/fifo/ticketing/domain/performance/dto/AdminPerformanceBookDetailDto.kt
package com.fifo.ticketing.domain.performance.dto

class AdminPerformanceBookDetailDto(
    val id: Long?,
    val title: String?,
    val encodedFileName: String?,
    val totalPrice: Long?,
    val totalQuantity: Long?,
) {
    var urlPrefix: String? = null

    val url: String
        get() = urlPrefix + encodedFileName
}